### ISSUE TITLE
The tests are not running anymore after the sidebar has closed

### ISSUE THREAD

**Issue by MatonAnthony** [mdn/doc-linter-webextension#7]
**Current behavior:** 
Once the sidebar has been closed once, the tests are not running when the sidebar is reopened.

**Expected behavior:**  
The test should be able to run whenever the sidebar has been opened independently of any previous state.

**Step to reproduce:**
1. Open an MDN page in edit mode
2. Launch the test suite
3. Close the sidebar
4. Reopen the sidebar
5. Relaunch the test suite

**Debug information:**
```log
TypeError: this.port is undefined panel.js:49:3
```
is triggered by this action.

I think this is triggerred because the connection is initialized by the content_script when it's loaded.
```javascript
const port = browser.runtime.connect(browser.runtime.id);
```
And so when I close the sidebar, the sidebar is unloaded and when I reopen it he doesn't have a connection anymore since the content_script isn't reloaded.

**Comment by MatonAnthony** [comment]
This seems to work with 807de2f thanks to @wbamberg for the explanations about the lifecycle of the differents parts of the Webextension and his idea about using a background script.
